File: ComPDFKit for Windows/Examples/Samples/ImageExtractTest/CS/ImageExtractTest.cs
using ComPDFKit.PDFDocument;
using System;
using System.IO;

namespace ImageExtractTest
{
    internal class ImageExtractTest
    {
        static private string outputPath = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory()))) + "\\Output\\CS";

        static void Main(string[] args)
        {
            #region Perparation work
            Console.WriteLine("Running Bookmark test sample…\r\n");

            SDKLicenseHelper.LicenseVerify();

            CPDFDocument document = CPDFDocument.InitWithFilePath("CommonFivePage.pdf");

            if (!Directory.Exists(outputPath))
            {
                Directory.CreateDirectory(outputPath);
            }

            #endregion

            #region Sample 1: Extract image

            ExtractImage(document);
            document.Release();
            Console.WriteLine("--------------------");

            #endregion

            Console.WriteLine("Done");
            Console.WriteLine("--------------------");
            Console.ReadLine();
        }

        /// <summary>
        /// Extract all images from document
        /// </summary>
        /// <param name="document">Regular documet with some picture</param>
        static private void ExtractImage(CPDFDocument document)
        { 
            document.ExtractImage("1-5", outputPath);
        }
    }
}

File: ComPDFKit for Windows/Examples/Samples/ImageExtractTest/VB/ImageExtractTest.vb
Imports ComPDFKit.PDFDocument
Imports System
Imports System.IO

Module ImageExtractTest
    Private outputPath As String = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory()))) & "\Output\VB"

    Sub Main()
        ' Perparation work
        Console.WriteLine("Running Bookmark test sample..." & vbCrLf)

        SDKLicenseHelper.LicenseVerify()

        Dim document As CPDFDocument = CPDFDocument.InitWithFilePath("CommonFivePage.pdf")

        If Not Directory.Exists(outputPath) Then
            Directory.CreateDirectory(outputPath)
        End If

        ' Sample 1: Extract image
        ExtractImage(document)
        document.Release()
        Console.WriteLine("--------------------")

        Console.WriteLine("Done")
        Console.WriteLine("--------------------")
        Console.ReadLine()
    End Sub

    ' Extract all images from document
    Private Sub ExtractImage(document As CPDFDocument)
        document.ExtractImage("1-5", outputPath)
    End Sub
End Module
